### PR TITLE
Include working coverage script

### LINKED ISSUE #227: Fix code coverage reporting
IIRC, I _think_ the problem was that remap-istanbul was trying to load some files that weren't on disk (their content had been bundled, making inclusion unnecessary, but it still caused a problem even though the content existed in the sourcemap's `sourcesContent` array), but I haven't thoroughly investigated yet.

### PR DESCRIPTION
**Pull request by lukastaegert**
This PR contains:
- [ ] bugfix
- [ ] feature
- [ ] refactor
- [ ] documentation
- [x] other

Are tests included?
- [ ] yes (*bugfixes and features will not be merged without tests*)
- [x] no

Breaking Changes?
- [ ] yes (*breaking changes will not be merged unless absolutely necessary*)
- [x] no

List any relevant issue numbers:
Resolves #227 

### Description
This includes a script to run tests with coverage.
TODO:
- [x] Run via CI
- [x] Integrate with some Github service for PRs

### CODE REVIEW

**Comment by TrySound** [comment]
Are you gonna bundle babel with rollup? How big bundle with this change?

**Comment by lukastaegert** [comment]
Babel is definitely not bundled and it is only a transitive devDependency via istanbul-lib-instrument and nyc anway. There should be not change for users of Rollup, both in Rollup's install size and the size of generated bundles. We should just make sure that we do not accidentally publish the instrumented Rollup bundle.

**Comment by lukastaegert** [comment]
So to make it clear, this is only for Rollup's own coverage. If Rollup consumers need coverage, they need to set up something similar to what has been done in this PR.

**Comment by TrySound** [comment]
Ah, great! Thanks for answer

**Comment by shellscape** [comment]
This addresses https://github.com/rollup/rollup/issues/227. That issue is worth reviewing as there's some good info in it.

**Comment by lukastaegert** [comment]
@shellscape Since this is working really well now, I'll put it on master to be able to have coverage diff with master for other PRs. In the end it turns out the special instrumentation plugin was not really necessary as nyc is working really well with sourcemaps now, all we needed were additional sourcemaps for bin/rollup and a proper include/exclude .nycrc file. We could consider adding a coverage limit for PRs in the future though for now I'm happy with this being a tool for reference to spot untested code paths.

File: package.json [hunk 1]
```diff
@@ -11,12 +11,11 @@
...
   "scripts": {
     "build": "shx rm -rf dist && git rev-parse HEAD > .commithash && rollup -c && shx cp src/rollup/types.d.ts dist/rollup.d.ts && shx chmod a+x bin/rollup",
-    "build:test": "shx rm -rf dist && rollup -c --configNoBrowser && shx cp src/rollup/types.d.ts dist/rollup.d.ts && shx chmod a+x bin/rollup",
+    "build:test": "shx rm -rf dist && rollup -c --configTest && shx cp src/rollup/types.d.ts dist/rollup.d.ts && shx chmod a+x bin/rollup",
     "build:bootstrap": "bin/rollup -c && shx cp src/rollup/types.d.ts dist/rollup.d.ts && shx chmod a+x bin/rollup",
-    "ci:coverage": "npm run test:coverage",
     "ci:lint": "npm run lint:nofix && npm run security",
     "ci:test": "npm run build:test && npm run build:bootstrap && npm run test:all",
-    "ci:test_12": "npm run build:test && npm run build:bootstrap && npm run test:only && npm run test:typescript",
+    "ci:coverage": "npm run build:test && nyc --reporter lcovonly mocha && codecov",
```
Comment (shellscape) [inline package.json]: _(late to the party, but I'll submit a PR for this)_

We can eliminate the `codecov` dependency by adding this to the circle ci config on the last node version test:

```yml
      - run:
          name: Submit coverage data to codecov.
          command: bash <(curl -s https://codecov.io/bash)
          when: on_success
```

**Pull request merged**